src/OrbitGl/CaptureViewElement.cpp

```diff
@@ -248,4 +247,49 @@ void CaptureViewElement::RequestUpdate(RequestUpdateScope scope) {
   }
 }
 
+void CaptureViewElement::PreRender(PrimitiveAssembler& primitive_assembler,
+                                   TextRenderer& text_renderer) {
+  primitive_assembler.PushTranslation(0, 0, DetermineZOffset());
+  text_renderer.PushTranslation(0, 0, DetermineZOffset());
+
+  if (RequestSeparateRenderGroup()) {
+    static const std::string kGroupPrefix = "|rg_cve_";
+    BatchRenderGroupStateManager* manager = primitive_assembler.GetRenderGroupManager();
+
+    previous_batcher_render_group_name_ = primitive_assembler.GetCurrentRenderGroupName();
+    const std::string new_batcher_render_group_name =
+        previous_batcher_render_group_name_ + kGroupPrefix + std::to_string(GetUid());
+    primitive_assembler.SetCurrentRenderGroupName(new_batcher_render_group_name);
+
+    BatchRenderGroupState state = manager->GetGroupState(previous_batcher_render_group_name_);
+    state.stencil.pos = {GetPos()[0], GetPos()[1]};
+    state.stencil.size = {GetSize()[0], GetSize()[1]};
+    state.stencil.enabled = true;
+    manager->SetGroupState(new_batcher_render_group_name, state);
+
+    previous_text_render_group_name_ = text_renderer.GetCurrentRenderGroupName();
+    std::string new_text_render_group_name =
+        previous_text_render_group_name_ + kGroupPrefix + std::to_string(GetUid());
```

Comment:
Same as above.

Also nit: `new_batcher_render_group_name` is declared `const`, but `new_text_render_group_name` is not. Also adding `const` here would be a nice touch.